3523/1/top_gems.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require 'open-uri'
+require 'nokogiri'
+require 'mechanize'
+require 'json'
+require_relative 'linker'
+require_relative 'githubinfo'
+require_relative 'sorter'
+require_relative 'filereader'
+
+options = {}
+
+OptionParser.new do |opts|
+  opts.banner = 'options'
+
+  opts.on('--top=NUM') do |num|
+    options[:top] = num.to_i
+  end
+
+  opts.on('--name=NAME') do |name|
+    options[:name] = name
+  end
+
+  opts.on('--file=PATH') do |path|
+    options[:gems] = YAML.load_file(path)['gems']
+  end
+end.parse!
+
+filereader = FileReader.new('gems.yaml')
+linker = Linker.new(filereader.read)
+githubinfo = GitHubInfo.new(linker.find_links)
+sorter = Sorter.new(githubinfo.info)
+
```

Comment:
If you change class name and method call order it'll became infinity times more readable
```
gem_names = FileReader.new('gems.yaml').read
links = Linker.new(gem_names).link
gems_info = GitHubInfo.new(linker.find_links).info
list = InfoList.new(gems_info)
```